src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -359,14 +358,19 @@ public function initialized($id)
     public function getServiceIds()
     {
         $ids = array();
+        $reversedMethodMap = array_change_key_case(array_flip($this->methodMap), \CASE_LOWER);
```

Comment:
You're right, i didnt consider `PhpDumper::generateMethodName` on this, which shows method names are generated from id.

However, they can get suffixed due duplicate methods, but that doesnt mean the id should change as well (this is the key in `methodMap` which can be perfectly unique).

I covered exactly that in [tests](https://github.com/symfony/symfony/pull/19690/files#diff-8aeba2b88d0347062017ee160a6beb49R619) (imagine a suffixed/generated method though).